dali/python/nvidia/dali/_utils/external_source_impl.py

```diff
@@ -65,6 +65,10 @@ def __str__(self) -> str:
     "Unsupported callback return type. Expected NumPy array, PyTorch or MXNet cpu tensors, "
     "DALI TensorCPU, list of those types or DALI TensorListCPU representing batch. Got `{}` instead.")
 
+_tf_uniform_error_msg = (
+    "Unsupported callback return value. TensorFlow requires that the batches produced by input "
+    "datasets or ExternalSource `source` callback in batch mode (that is when batch=True) "
+    " are dense and uniform - this means that every sample has the same shape. Got `{}` instead.")
```

Comment:
```suggestion
    " are dense and uniform - this means that every sample has the same shape and type. Got `{}` instead.")
```

Reply:
Hmm, I don't think non-uniform types would work ever in anything. I'm not sure it's worth mentioning.

Reply:
Ok